stream.go

```diff
@@ -0,0 +1,510 @@
+
+// Sfixed32Packed writes a slice of values of proto type sfixed32 to the stream,
+// in packed form.
+func (ps *ProtoStream) Sfixed32Packed(fieldNumber int, values []int32) error {
+	if len(values) == 0 {
+		return nil
+	}
+	ps.scratchBuffer = ps.scratchBuffer[:0]
+	for _, value := range values {
+		ps.scratchBuffer = protowire.AppendFixed32(ps.scratchBuffer, uint32(value))
+	}
+	return ps.writeScratchAsPacked(fieldNumber)
+}
+
+// Sfixed64 writes a value of proto type sfixed64 to the stream.
+func (ps *ProtoStream) Sfixed64(fieldNumber int, value int64) error {
+	if value == 0 {
+		return nil
+	}
+	ps.scratchBuffer = ps.scratchBuffer[:0]
+	ps.encodeKeyToScratch(fieldNumber, wt64Bit)
+	ps.scratchBuffer = protowire.AppendFixed64(ps.scratchBuffer, uint64(value))
+	return ps.writeScratch()
+}
+
+// Sfixed64Packed writes a slice of values of proto type sfixed64 to the stream,
+// in packed form.
+func (ps *ProtoStream) Sfixed64Packed(fieldNumber int, values []int64) error {
+	if len(values) == 0 {
+		return nil
+	}
+	ps.scratchBuffer = ps.scratchBuffer[:0]
+	for _, value := range values {
+		ps.scratchBuffer = protowire.AppendFixed64(ps.scratchBuffer, uint64(value))
+	}
+	return ps.writeScratchAsPacked(fieldNumber)
+}
+
+// Bool writes a value of proto type bool to the stream.
+func (ps *ProtoStream) Bool(fieldNumber int, value bool) error {
+	if value == false {
+		return nil
+	}
+	ps.scratchBuffer = ps.scratchBuffer[:0]
+	ps.encodeKeyToScratch(fieldNumber, wtVarint)
+	var bit uint64
+	if value {
+		bit = 1
+	}
+	ps.scratchBuffer = protowire.AppendVarint(ps.scratchBuffer, bit)
+	return ps.writeScratch()
+}
+
+// String writes a string to the stream.
+func (ps *ProtoStream) String(fieldNumber int, value string) error {
+	if len(value) == 0 {
+		return nil
+	}
+	ps.scratchBuffer = ps.scratchBuffer[:0]
+	ps.encodeKeyToScratch(fieldNumber, wtLengthDelimited)
+	ps.scratchBuffer = protowire.AppendVarint(ps.scratchBuffer, uint64(len(value)))
+	err := ps.writeScratch()
+	if err != nil {
+		return err
+	}
+
+	return ps.writeAllString(value)
+}
+
+// Bytes writes the given bytes to the stream.
+func (ps *ProtoStream) Bytes(fieldNumber int, value []byte) error {
+	if len(value) == 0 {
+		return nil
+	}
+	ps.scratchBuffer = ps.scratchBuffer[:0]
+	ps.encodeKeyToScratch(fieldNumber, wtLengthDelimited)
+	ps.scratchBuffer = protowire.AppendVarint(ps.scratchBuffer, uint64(len(value)))
+	err := ps.writeScratch()
+	if err != nil {
+		return err
+	}
+
+	return ps.writeAll(value)
+}
+
+// Embedded is used for constructing embedded messages.  It calls the given
+// function with a new ProtoStream, then embeds the result in the current
+// stream.
+//
+// NOTE: if the inner function creates an empty message (such as for a struct
+// at its zero value), that empty message will still be added to the stream.
+func (ps *ProtoStream) Embedded(fieldNumber int, inner func(*ProtoStream) error) error {
+	// create a new child, writing to a buffer, if one does not already exist
```

Comment:
Sorry super nit, but do you mind capitalizing comments and ending with `.`

Reply:
You resolved this but didn't fix it :P

Reply:
Oops!  I thought I did!  Sorry!

Reply:
It says "Outdated" now so I think I got it :)